tests/Mocks/Validation/AddressMock.php

```diff
@@ -55,9 +55,34 @@ public function mockValidationSuccess(): ShipEngine
         return new ShipEngine("", $addressFormatter, $mockOptions);
     }
 
-    public function mockValidationFailure(): ShipEngine
+    public function mockInvalidErrorFromShipEngine(): ShipEngine
     {
-        $mock = new MockHandler([new ClientException("Insufficient or Incorrect Address Data", new Request('POST', 'https://api.shipengine.com/v1/addresses/validate'))]);
+        $response = [
+            'request_id' => 'c57869c1-a123-444e-80da-a7e7fedf4c27',
+            'errors' => [
+                [
+                    'error_source' => 'shipengine',
+                    'error_type' => 'security',
+                    'error_code' => 'unauthorized',
+                    'message' => 'The API key is invalid. Please see https://www.shipengine.com/docs/auth'
+                ]
+            ]
+        ];
+
+        $mock = new MockHandler([new Response(200, [], json_encode($response))]);
+        $mockStack = HandlerStack::create($mock);
+        $mockOptions = ['handler' => $mockStack];
+
+        $addressFormatter = new ArrayFormatter();
+
+        return new ShipEngine("", $addressFormatter, $mockOptions);
+    }
+
+    public function mockInvalidConnectionToShipEngine(): ShipEngine
```

Comment:
I kinda see why this is an invalid connection. But this is an exception from shipengine, because you hit their own validation. Unsure about the name. But, not worth using the brain power to come up with another one.

Reply:
We spoke in person, basically tldr - Trying to find a good map for a null response from cURL and mapping to an exception in Guzzle